packages/terminal/src/widget.ts

```diff
@@ -176,8 +177,10 @@ export class Terminal extends Widget implements ITerminal.ITerminal {
         });
       }
     }
-    this._term.dispose();
-    super.dispose();
+    if (this.getOption('disposeOnExit')) {
+      this._term.dispose();
+      super.dispose();
+    }
```

Comment:
This is not the right place to do that because it will skip clearing the terminal resources.

This should be handle on line 74 when reacting to `session.disposed`.